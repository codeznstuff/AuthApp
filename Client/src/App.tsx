import React from 'react';
import { useAuthorization } from './context/Authorization/AuthorizationContext';
import AuthenticatedApp from './AuthenticatedApp';
import UnauthenticatedApp from './UnauthenticatedApp';
import { IApplicationUser } from './services/AuthorizationApi';

const App = () => {
  const user: IApplicationUser = useAuthorization();
  return user.userId ? <AuthenticatedApp /> : <UnauthenticatedApp />;
};

export default App;
