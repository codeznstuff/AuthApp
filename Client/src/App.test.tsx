import React from 'react';
import { render, RenderResult, cleanup } from '@testing-library/react';
import App from './App';

describe('App', () => {
  let app: RenderResult;

  beforeEach(() => {
    app = render(<App />);
  });

  afterEach(cleanup);

  it('matches the initial snapshot', () => {
    expect(app.container.firstChild).toMatchSnapshot();
  });
});
