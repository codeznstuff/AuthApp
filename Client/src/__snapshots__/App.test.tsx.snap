// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches the initial snapshot 1`] = `
<div
  class="container"
  id="outer"
  style="width: 100%; height: 100vh; display: flex;"
>
  <div
    id="inner"
    style="margin: auto;"
  >
    <h1
      class="display-2 text-center"
    >
      Sorry
    </h1>
    <p
      class="lead text-center"
    >
      Looks like something went wrong on our end.
    </p>
    <p
      class="lead text-center"
    >
      Please check your network connection and try back later.
    </p>
  </div>
</div>
`;
